server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -199,41 +208,40 @@ private void sendJobMessage(JobMessage jobMessage) throws ActiveMQException, Jso
      * Builds an AsyncJobStatus instance from the given job details. The job status will not be
      * a managed entity and will need to be manually persisted by the caller.
      *
-     * @param detail
-     *  The job detail to use to build the job status
+     * @param builder
+     *  The JobBuilder to use to build the job status
      *
      * @throws IllegalArgumentException
      *  if detail is null
      *
      * @return
      *  the newly constructed, unmanaged AsyncJobStatus instance
      */
-    private AsyncJobStatus buildJobStatus(JobDetail detail) {
-        if (detail == null) {
-            throw new IllegalArgumentException("job detail is null");
+    private AsyncJobStatus buildJobStatus(JobBuilder builder) {
+        if (builder == null) {
+            throw new IllegalArgumentException("job builder is null");
         }
 
-        // TODO: Rewrite this method to use the upcoming JobBuilder
-
         AsyncJobStatus job = new AsyncJobStatus();
 
-        job.setJobClass(detail.getJobClass());
-        job.setName(detail.getKey().getName());
-        job.setGroup(detail.getKey().getGroup());
+        job.setJobClass(builder.getJobClass());
```

Comment:
We should store a job key instead of the actual job class so that we have flexibility to change the class name if we choose -- storing the actual class name has bitten us with Quartz in the past. When receiving a job message, candlepin should look up the Job class via some sort of registry / config.

Reply:
Done. The way I've sorted this for the moment is adding a static map to JobManager which can be updated via calls to the static methods registerJob and unregisterJob, and keys can be resolved by calling getJobClass. The idea for the fix is that actual job classes will have a static block that looks like this:

```
public class RefreshPoolsJob {
    static {
        JobManager.registerJob("refresh_pools", RefreshPoolsJob.class);
    }

    ...
}
```

Reply:
:+1: